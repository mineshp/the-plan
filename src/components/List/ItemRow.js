import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';


const ItemRow = ({ data, listID }) => (
    <Table.Row>
        <Table.Cell>{data.name}</Table.Cell>
        <Table.Cell>{data.createdDate}</Table.Cell>
        <Table.Cell>{data.updatedDate}</Table.Cell>
        <Table.Cell>{data.notes}</Table.Cell>
        <Table.Cell>{data.owner}</Table.Cell>
        <Table.Cell><Button as='a' color='teal' href={`/list/${listID}/item/${data.id}/delete`}>Delete</Button></Table.Cell>
    </Table.Row>
);

ItemRow.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    createdDate: PropTypes.string.isRequired,
    updatedDate: PropTypes.string.isRequired
    }).isRequired
}

export default ItemRow;