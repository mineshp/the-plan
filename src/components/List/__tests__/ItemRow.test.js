import React from 'react';
import ItemRow from '../ItemRow';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from 'semantic-ui-react';

const mockRowData = {
    id: '1',
    name: 'test',
    createdDate: "2017-08-25T20:22:40.994Z",
    updatedDate: "2017-08-29T12:09:29.443Z",
    notes: 'I am a note',
    owner: 'Miss Appleby'
};

describe('ItemRow', () => {
    it('renders item row correctly', () => {
        const tree = renderer.create(
            <ItemRow data={mockRowData} listID='123' />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a delete button', () => {
        const wrapper = shallow(<ItemRow data={mockRowData} listID='123' />);
        const DeleteBtn = wrapper.find(Button);

        expect(DeleteBtn.length).toBe(1);
        expect(DeleteBtn.props().children).toBe('Delete');
        expect(DeleteBtn.props().href).toBe('/list/123/item/1/delete');
    });
});