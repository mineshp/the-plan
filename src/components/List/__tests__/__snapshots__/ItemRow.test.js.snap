// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRow renders item row correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    test
  </td>
  <td
    className=""
  >
    2017-08-25T20:22:40.994Z
  </td>
  <td
    className=""
  >
    2017-08-29T12:09:29.443Z
  </td>
  <td
    className=""
  >
    I am a note
  </td>
  <td
    className=""
  >
    Miss Appleby
  </td>
  <td
    className=""
  >
    <a
      className="ui teal button"
      disabled={undefined}
      href="/list/123/item/1/delete"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Delete
    </a>
  </td>
</tr>
`;
